ahds/data_stream.py

```diff
@@ -55,124 +161,59 @@ def byterle_decoder(data, output_size):
-    """Decode HxRLE data stream
-    
-    If C-extension is not compiled it will use a (slower) Python equivalent
+"""Decode HxRLE data stream
+   
+   func:: hxbyterle_decode
+   if C-extension is not compiled it will use a (slower) Python equivalent
     
-    :param int output_size: the number of items when ``data`` is uncompressed
-    :param str data: a raw stream of data to be unpacked
-    :return numpy.array output: an array of ``numpy.uint8``
-    """
-    output = byterle_decoder(data, output_size)
-    assert len(output) == output_size     
-    return output
+   :param int output_size: the number of items when ``data`` is uncompressed
+   :param str data: a raw stream of data to be unpacked
+   :return np.array output: an array of ``np.uint8``
+"""
+hxbyterle_decode = byterle_decoder
 
 
-def hxzip_decode(data_size, data):
+def hxzip_decode(data,dtype=None,count = 0):
     """Decode HxZip data stream
     
     :param int data_size: the number of items when ``data`` is uncompressed
     :param str data: a raw stream of data to be unpacked
-    :return numpy.array output: an array of ``numpy.uint8``
-    """
-    import zlib
-    data_stream = zlib.decompress(data)
-    output = numpy.array(struct.unpack('<{}B'.format(len(data_stream)), data_stream), dtype=numpy.uint8)
-    assert len(output) == data_size
-    return output
-
-
-def unpack_binary(data_pointer, definitions, data):
-    """Unpack binary data using ``struct.unpack``
-    
-    :param data_pointer: metadata for the ``data_pointer`` attribute for this data stream
-    :type data_pointer: ``ahds.header.Block``
-    :param definitions: definitions specified in the header
-    :type definitions: ``ahds.header.Block``
-    :param bytes data: raw binary data to be unpacked
-    :return tuple output: unpacked data 
-    """
-    
-    if data_pointer.data_dimension:
-        data_dimension = data_pointer.data_dimension
-    else:
-        data_dimension = 1 # if data_dimension is None
-        
-    if data_pointer.data_type == "float":
-        data_type = "f" * data_dimension
-    elif data_pointer.data_type == "int":
-        data_type = "i" * data_dimension # assume signed int
-    elif data_pointer.data_type == "byte":
-        data_type = "b" * data_dimension # assume signed char
-
-    # get this streams size from the definitions
-    try: 
-        data_length = int(getattr(definitions, data_pointer.data_name))
-    except AttributeError:
-        # quickfix
-        """
-        :TODO: nNodes definition fix
-        """
-        try:
-            data_length = int(getattr(definitions, 'Nodes'))
-        except AttributeError:
-            x, y, z = definitions.Lattice
-            data_length = x * y * z
-        
-    output = numpy.array(struct.unpack('<' + '{}'.format(data_type) * data_length, data)) # assume little-endian
-    output = output.reshape(data_length, data_dimension)
-    return output
-
-
-def unpack_ascii(data):
```

Comment:
Missing.

Reply:
I think i replaced it by numpy.fromstring for the same reason as for unpack_binary.